WeslToml.md

```diff
@@ -0,0 +1,135 @@
+# We watch this directory for changes.
+root = "./shaders"
+
+# Optional
+include = [ "shaders/**/*.wesl", "shaders/**/*.wgsl" ]
+
+# Optional.
+# Some projects have large folders that we shouldn't react to. 
+exclude = [ "**/test" ]
+
+# Optional, can be auto-inferred from the existence of a package.json.
+# Necessary when both a `package.json` and a `Cargo.toml` are present.
+# Inclusion of this field is encouraged.
+package-manager = "npm"
+
+# Lists all used dependencies
+[dependencies]
+# Shorthand for `foolib = { package = "foolib" }`
+foolib = {}
+ # Can be used for renaming packages. Now bevy in my code is called "cute_bevy".
+cute_bevy = { package = "bevy" }
+# File path to a folder with a wesl.toml. Simplest kind of dependency.
+mylib = { path = "../mylib" }
+```
+
+
+[^1]: Linkers are free to offer their own configuration, in addition to supporting the `wesl.toml` format.
+For example, wesl-js in the browser would be configured via a roughly equivalent JSON object. This lets wesl-js avoid bundling a whole toml file format parser.
+
+
+## Semantics
+
+A `wesl.toml` file is mandatory for libraries.
+
+For user applications, feel free to start without a `wesl.toml`. We'll assume reasonable defaults, which will evolve over time.
+
+As your application grows, we suggest adding a `wesl.toml` for better consistency. This lets you pin down the exact behaviour of wesl.
+
+### `edition` field
+
+Specifies which edition is used by wesl. All wesl editions can be used.
+
+- Mandatory
+  - Falls back to latest edition that tools know.
+- Currently only `unstable_2025` is accepted, since we do not have an edition yet.
+  - No incompatible changes are planned, but we have not committed to stable version for long term support.
+
+### `package-manager` field
+
+Which package manager is used for resolving wesl libraries.
+
+- Optional, but encouraged.
+  - Can be inferred from the existence of certain files (`package.json` and `Cargo.toml`)
+- `npm` and `cargo are accepted.
+
+It is limited to one package manager to reduce implementation complexity.
+We do not have any wesl implementations that would allow for mixing and matching packages from different package managers.
+For dual publishing, the expectation is that one would have a primary package manager, and then attempt to mirror the structure for other package managers.
+
+## `include` and `exclude` fields
+
+Specifies an array of patterns where wesl files are located. They patterns are relative to the `wesl.toml` file.
+
+The default value of `include` is all wesl and wgsl files in the `root` directory, recursively.
+The default value of `exclude` is an empty array.
+
+The patterns are glob patterns, which support
+- `*` for zero or more characters in file/folder names
+- `?` for one character in a file/folder name
+- `**/` for any directory nested to any level
+
+The globs must match file names, and the last part of the patterns must contain a file extension.
```

Comment:
Wait, how should the globs work? Should we allow `include = ["shaders"]`? If yes, what are the semantics? Does that include `.wgsl` files or does it only include `.wesl` files?